File: benchmarks/pandas/bench_cummax_cummin_str.py
"""
Benchmark: Series.cummax() / cummin() on string Series of 10k elements.
Outputs JSON: {"function": "cummax_cummin_str", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time
import pandas as pd

SIZE = 10_000
WARMUP = 5
ITERATIONS = 50

words = ["apple", "banana", "cherry", "date", "elderberry", "fig", "grape", "honeydew"]
data = [words[i % len(words)] for i in range(SIZE)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.cummax()
    s.cummin()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.cummax()
    s.cummin()
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "cummax_cummin_str", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_cummax_cummin_str.ts
/**
 * Benchmark: cummax / cummin on string Series of 10k elements.
 * Outputs JSON: {"function": "cummax_cummin_str", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series, cummax, cummin } from "../../src/index.ts";

const SIZE = 10_000;
const WARMUP = 5;
const ITERATIONS = 50;

const words = ["apple", "banana", "cherry", "date", "elderberry", "fig", "grape", "honeydew"];
const data: string[] = Array.from({ length: SIZE }, (_, i) => words[i % words.length]);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  cummax(s);
  cummin(s);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  cummax(s);
  cummin(s);
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "cummax_cummin_str", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));
